openedx/features/enterprise_support/utils.py

```diff
@@ -283,6 +283,23 @@ def _get_sync_learner_profile_data(enterprise_customer):
     return False
 
 
+def get_enterprise_learner_portals(user):
+    """
+    Gets the formatted portal names and slugs that can be used
+    to generate links for enabled enterprise Learner Portals
+    """
+    # Prevent a circular import.
+    from openedx.features.enterprise_support.api import get_enterprise_learner_data, enterprise_enabled
+
+    return [{
+        'name': learner_data.get('enterprise_customer', {}).get('name'),
+        'slug': learner_data.get('enterprise_customer', {}).get('branding_configuration', {}).get('enterprise_slug'),
+        'logo': learner_data.get('enterprise_customer', {}).get('branding_configuration', {}).get('logo'),
+    } for learner_data in get_enterprise_learner_data(user)
+        if learner_data.get('enterprise_customer', {}).get('enable_learner_portal')
+    ] if enterprise_enabled() and is_enterprise_learner(user) else None
```

Comment:
2 potential scaling concerns as we roll this out. I haven't ran this locally to profile if these are true, but highlighting as potential areas of concern:
- `get_enterprise_learner_data` uses the `EnterpriseApiClient` to fetch the data, which means the LMS is making a rest call to another endpoint in the LMS. I believe may result in rate limiting ourselves since we'll be doing this on every header call. Even if we don't rate limit, we'll still be serializing unnecessarily. I wonder if we should refactor this code to not use the API client in order to support this.

- Lesser concern, but we may need to introduce some sort of caching around `is_enterprise_learner` Again everytime the header is loaded this would get called (every page load) which does a DB lookup on EnterpriseCustomerUser: https://github.com/edx/edx-platform/blob/ec2369d88fffa0cd2199f0129c64605d4475b1e2/openedx/features/enterprise_support/utils.py#L319

Reply:
Ahh great call, I'll look at the caching utils for the is_enterprise_learner call, though it's possible it's a better approach to just check the JWT here as well in python.

For the enterprise learner data I'll see if I can just get that data directly from the source, no reason not to. Looking back, a bit odd to me that this exists at all in here but I'm sure there are reasons. Again can visit user specific caching with some short TTL here

Reply:
Can do something similar to the following and will have all the data I need without proxying through the API. Will look into caching this for a short period next.
```
EnterpriseCustomerUser.objects.filter(
        user_id=user.id, enterprise_customer__enable_learner_portal=True
    ).prefetch_related(
        'enterprise_customer', 'enterprise_customer__branding_configuration'
    ).order_by('-enterprise_customer__active', '-modified')
```

Reply:
Yup, that seems right and would be much more performant than going through the API. For historical context the original reason we set it up this way was with the idea we would/could move the enterprise app outside of the LMS context.

Reply:
Turns out we have a bunch of caching utils we could use, `@request_cached`, `@process_cached`, and `TieredCache` but none of those really fit our case. Talked to @iloveagent57 and he recommended just using the django cache with a short ttl set, so I'll get that up and in here, thinking just a naive ~5/10 minute TTL should alleviate a bulk of the load without messing with the experience for our enterprise learners too much

Reply:
Rather than a generic cache, another option would put it in the users session. We do that here for the users enterprise so adding the branding config seems logical: https://github.com/edx/edx-enterprise/blob/a50bdd773fa416e6bc2cb579f7f603c551dd8e9e/enterprise/models.py#L908